Show closed accounts on mobile UI (#4581)

alcroito opened the pull request:
This PR adds an additional 'Closed Accounts' group to the accounts page, below the Budget and Off Budget groups.
This allows filtering / searching for transactions in closed accounts.

alcroito commented:
> It would be nice if the closed accounts are collapsed by default similar to how closed accounts are in desktop

I modified the code to collapse the accounts using the same local preference that's used for the desktop UI.

joel-jeremy commented:
> > It would be nice if the closed accounts are collapsed by default similar to how closed accounts are in desktop
> 
> I modified the code to collapse the accounts using the same local preference that's used for the desktop UI.

Thanks! This looks good. One last nitpick - can we keep the chevron instead of the ...?

joel-jeremy commented on packages/desktop-client/src/components/mobile/accounts/Accounts.tsx:
```diff
@@ -247,13 +261,43 @@ function AllAccountList({
+    // might trigger before that.
+    if (shouldScrollToClosedAccounts && showClosedAccounts) {
+      setShouldScrollToClosedAccounts(false);
+      setTimeout(() => {
```
Can we just call this directly from `onToggleClosedAccounts`? That should remove the need for `shouldScrollToClosedAccounts` state

alcroito replied:
That's how I originally had it, but then the CodeRabbit AI suggested the current approach, and I also considered that this is more 'clean' because calling scroll does sound like something that should be done in `useEffect()`.
I can restore the original code though.

alcroito replied:
Removed the `useEffect`, as requested.

joel-jeremy commented on packages/desktop-client/src/components/mobile/accounts/Accounts.tsx:
```diff
@@ -307,6 +351,27 @@ function AllAccountList({
+              accounts={closedAccounts}
+              getAccountBalance={getAccountBalance}
+              onOpenAccount={onOpenAccount}
+              ref={el => {
+                if (el) closedAccountsRef.current = el;
+              }}
```
```suggestion
              ref={closedAccountsRef}
```

alcroito replied:
Keeping this as-is, because it doesn't work correctly otherwise.

joel-jeremy commented on packages/desktop-client/src/components/mobile/accounts/Accounts.tsx:
```diff
@@ -44,22 +49,28 @@ type AccountHeaderProps<SheetFieldName extends SheetFields<'account'>> = {
   name,
   amount,
   style = {},
+  pressHandler,
```
```suggestion
  onPress,
```

alcroito replied:
Done.

matt-fidd commented:
Thanks for this!

A couple of things I noticed from testing:
- This looks to be showing my closed accounts in the on/off budget sections too, not just the closed section.
- Could the chevron/arrow icon rotate when the closed accounts are open/closed?

alcroito commented:
> * This looks to be showing my closed accounts in the on/off budget sections too, not just the closed section.

Oh, how embarrassing. Fixed in latest commit.

> * Could the chevron/arrow icon rotate when the closed accounts are open/closed?

Added. The right cheveron is shown when the closed accounts are not shown, and the down cheveron is shown when the closed accounts are shown. Basically a 90 degree rotation.

matt-fidd reviewed (approved):
Great! Looks good to me, will leave for a day or so in case Jay has any more comments to add too.

joel-jeremy reviewed (approved):
LGTM!
